### PR TITLE
Implement the dayforce connector

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Implement the initial version of the dayforce.
Related to:- https://github.com/ballerina-platform/ballerina-library/issues/6552

## Approach
This is a connector generated by the OpenAPI specification. And did some modifications to the APIs that are returning paginated data.

### CODE REVIEW

**Review by ayeshLK — COMMENTED** [review]
Why do we have two client definitions ?
- ballerina/client.bal
- ballerina/modules/oas/client.bal

**Comment by Bhashinee** [comment]
> Why do we have two client definitions ?
> 
> * ballerina/client.bal
> * ballerina/modules/oas/client.bal

ballerina/client.bal is a wrapper client as we directly can't use the generated one because of some limitations related to pagination. We have to write a little bit of code in some of the APIs to support that.

**Comment by ayeshLK** [comment]
> ballerina/client.bal is a wrapper client as we directly can't use the generated one because of some limitations related to pagination. We have to write a little bit of code in some of the APIs to support that.

In that case shall we add the `sanitations.md` to the same PR ?

File: .github/workflows/build-with-bal-test-native.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: GraalVM Check
```
Comment (NipunaRanasinghe) [inline .github/workflows/build-with-bal-test-native.yml]: do we need this workflow? I think `build-with-bal-test-graalvm` will be enough

Comment (Bhashinee) [reply .github/workflows/build-with-bal-test-native.yml]: Removed with [fd44eef](https://github.com/ballerina-platform/module-ballerinax-dayforce/pull/1/commits/fd44eef26521315bd38534d2bae18858751504d8)

**Comment by Bhashinee** [comment]
> > ballerina/client.bal is a wrapper client as we directly can't use the generated one because of some limitations related to pagination. We have to write a little bit of code in some of the APIs to support that.
> 
> In that case shall we add the `sanitations.md` to the same PR ?

Added in [fd44eef](https://github.com/ballerina-platform/module-ballerinax-dayforce/pull/1/commits/fd44eef26521315bd38534d2bae18858751504d8)

File: ballerina/client.bal [hunk 2]
```diff
@@ -0,0 +1,3856 @@
...
+    # + return - Returns a page of employee data, `nil` if the given paginated data does not have a url(marking the end of the pages) or else an error if the request fails.
+    resource isolated function get [string clientNamespace]/v1/GetEmployeeBulkAPI/Data/[string jobId](Paging? pagination = ()) returns PaginatedPayload_IEnumerable_Employee|error? {
+        string:RegExp re = re `${self.serviceUrl}`;
+        if pagination !is () {
+            if pagination.Next is () || pagination.Next == "" {
+                return ();
+            }
+            string[] result = re.split(<string>pagination.Next);
+            string resourcePath = string `${result[1]}`;
+            PaginatedPayload_IEnumerable_Employee response = check self.paginatedClient->get(resourcePath);
```
Comment (NipunaRanasinghe) [inline ballerina/client.bal:93]: Why do we need a separate HTTP client to do this? Is there a possibility to reuse the same HTTP client we already have underneath the generated client?

Comment (Bhashinee) [reply ballerina/client.bal:93]: As we get an entire URL as the next page URL, what we need to do from the connector is, do a `get` call giving that URL. The generated client does not have a compatible resource for that.

File: .github/workflows/build-with-bal-test-native.yml (hunk 1 above)
Comment (NipunaRanasinghe) [reply .github/workflows/build-with-bal-test-native.yml]: @Bhashinee it seems you have removed the wrong workflow :)

Comment (Bhashinee) [reply .github/workflows/build-with-bal-test-native.yml]: Sorry, fixed in [1133d11](https://github.com/ballerina-platform/module-ballerinax-dayforce/pull/1/commits/1133d11a6f3cb1798ea4db7d256e967af867e151)

File: README.md [hunk 3]
```diff
@@ -1,2 +1,150 @@
...
+[![codecov](https://codecov.io/gh/ballerina-platform/module-ballerinax-dayforce/branch/main/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/module-ballerinax-dayforce)
+[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-dayforce.svg)](https://github.com/ballerina-platform/module-ballerinax-dayforce./commits/master)
+[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-dayforce/actions/workflows/build-with-bal-test-native.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-dayforce/actions/workflows/build-with-bal-test-native.yml)
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+## Overview
+
+Dayforce is a comprehensive human capital management system that covers the entire employee lifecycle including HR, payroll, benefits, talent management, workforce management, and services. The entire system resides on cloud that takes the burden of managing and replicating data on-premise.
+
+The Dayforce connector allows you to access the REST API of Dayforce HCM.
```
Comment (NipunaRanasinghe) [inline README.md]: Can we specify the specific REST API version that we are supporting

File: README.md [hunk 4]
```diff
@@ -1,2 +1,150 @@
...
+### Step 1: Create a Dayforce account
+
+Navigate to the [Dayforce](https://developers.dayforce.com/Build/Home.aspx) website and register. Follow the instructions to create an account. If your company has already purchased a namespace in Dayforce, use that to sign up. Otherwise, you can still use their `sample` environment by selecting `sample` option.
+
+### Step 2: Obtain the user credentials to access the Dayforce API
+
+To get the credentials for the `sample` environment, follow the instructions below:
+1. Navigate to API Explorer > Employee > GET Employees.
+2. Locate the basic authentication credentials, including the `username` and `password`, for the sample environment.
+
```
Comment (NipunaRanasinghe) [inline README.md]: Shall we add include screenshots to indicate the mentioned steps, wherever possible (similar to what we have done in other connectors)?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,150 @@
...
+   }, 
+   "https://www.dayforcehcm.com/Api");
+```
+
+### Step 3: Invoke the connector operation
+
+Now, utilize the available connector operations.
+```ballerina
+public function main() returns error? {
+    dayforce:Client dayforce = ...//
```
Comment (NipunaRanasinghe) [inline README.md]: it seems we have initialized 2 clients (one at the module level and one within the main function). Shall we correct this?

```suggestion
    dayforce:Client dayforce = ...//
```

File: ballerina/Ballerina.toml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+[package]
+distribution = "2201.8.0"
+org = "ballerinax"
+name = "dayforce"
+version = "0.1.0"
+authors = ["Ballerina"]
+repository = "https://github.com/ballerina-platform/module-ballerinax-dayforce"
+keywords = ["HCM", "Workforce Management", "HRIS"]
+#icon = "icon.png"
```
Comment (NipunaRanasinghe) [inline ballerina/Ballerina.toml]: Don't we have an icon for this? better to have one

File: gradle.properties [hunk 7]
```diff
@@ -0,0 +1,13 @@
+org.gradle.caching=true
+group=io.ballerina.stdlib
```
Comment (NipunaRanasinghe) [inline gradle.properties]: is this correct?
```suggestion
group=io.ballerina.lib
```

File: README.md (hunk 3 above)
Comment (Bhashinee) [reply README.md]: Fixed in [f54eb34](https://github.com/ballerina-platform/module-ballerinax-dayforce/pull/1/commits/f54eb34eb38de4c0777362f209ecad3871052ee2)

File: README.md (hunk 4 above)
Comment (Bhashinee) [reply README.md]: Fixed in [f54eb34](https://github.com/ballerina-platform/module-ballerinax-dayforce/pull/1/commits/f54eb34eb38de4c0777362f209ecad3871052ee2)

File: README.md (hunk 5 above)
Comment (Bhashinee) [reply README.md]: Fixed in [f54eb34](https://github.com/ballerina-platform/module-ballerinax-dayforce/pull/1/commits/f54eb34eb38de4c0777362f209ecad3871052ee2)

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**